### PR TITLE
Add date options

### PR DESCRIPTION
**Pull request by mattandrews**
Re-adds the year facet but tweaks it by modifying it into a date range. Also prepends some custom options (3/6 months ago). There's a TODO about making these dynamic but I couldn't think of a good way of deciding whether should show them – eg. we could hardcode a datestamp each time we do an import, we could make an additional query to check for the newest item in the database... or, we could strive to update this dataset at least once a quarter 😎 

Pairs with https://github.com/biglotteryfund/blf-alpha/pull/1400

### CODE REVIEW

File: grants/__snapshots__/search.test.js.snap [hunk 1]
```diff
@@ -4,7 +4,18 @@ exports[`Past Grants Search should return empty results for invalid queries 1`]
 Object {
   "facets": Object {
     "amountAwarded": Array [],
-    "awardDate": Array [],
+    "awardDate": Array [
+      Object {
+        "_id": "last3Months",
+        "label": "Last three months",
+        "value": "2018-07-25|2018-10-25",
```
Comment (mattandrews) [inline grants/__snapshots__/search.test.js.snap]: ... will this break the tests tomorrow? Not sure if this is something we can handle?

Comment (mattandrews) [reply grants/__snapshots__/search.test.js.snap]: ... yep: https://travis-ci.org/biglotteryfund/grants-service/builds/446084717?utm_medium=notification&utm_source=email

Comment (davidrapson) [reply grants/__snapshots__/search.test.js.snap]: Yeah you either have to delete or replace this value before taking a snapshot (like we do with the email tests in the main app). Or alternatively don't use a single snapshot. Maybe snapshot individual pieces where it makes sense but plain assert against the rest.

Comment (mattandrews) [reply grants/__snapshots__/search.test.js.snap]: Could you have a look again – I've just removed the snapshot comparison for that failing test in favour of just testing the number of results directly.

**Pull request merged**